lib/ecto/schema.ex

```diff
@@ -2061,6 +2062,30 @@ defmodule Ecto.Schema do
     Module.put_attribute(mod, :struct_fields, {name, assoc})
   end
 
+  defp validate_default!(_, nil), do: :ok
+  defp validate_default!(_, []), do: :ok
+
+  defp validate_default!({:parameterized, mod, _} = type, value) do 
```

Comment:
Do we need this clause? Ecto.Type will already do the same for us, no?

Reply:
You're correct! :) 
Removed it.